a11y-dialog.js

```diff
@@ -1,148 +1,308 @@
       focusableChildren[focusableChildren.length - 1].focus();
       event.preventDefault();
+    // If the SHIFT key is not being pressed (moving forwards) and the currently
+    // focused item is the last one, move the focus to the first focusable item
+    // from the dialog element
     } else if (!event.shiftKey && focusedItemIndex === focusableChildren.length - 1) {
       focusableChildren[0].focus();
       event.preventDefault();
     }
   }
 
-  // Helper function to focus first focusable item in node
-  function setFocusToFirstItem (node) {
-    var focusableChildren = getFocusableChildren(node);
-    if (focusableChildren.length) focusableChildren[0].focus();
+  // Retrieve siblings from given element
+  // @param {Element} node
+  // @return {Array<Element>}
+  function getSiblings (node) {
+    var siblings = toArray(node.parentNode.childNodes)
+    siblings.splice(siblings.indexOf(node), 1)
+    return siblings
   }
 
-  var focusedBeforeDialog;
+  // Instanciate a dialog
+  // @param {Element} node
+  // @param {(NodeList | Element | string)} targets
+  // @return {A11yDialog}
+  function A11yDialog (node, targets) {
+    // Prebind the functions that will be bound in addEventListener and
+    // removeEventListener to avoid losing references
+    this._show = this.show.bind(this);
+    this._hide = this.hide.bind(this);
+    this._maintainFocus = this._maintainFocus.bind(this);
+    this._bindKeypress = this._bindKeypress.bind(this);
+
+    // Keep a reference of the node on the instance
+    this.node = node;
 
-  /**
-   * A11yDialog constructor
-   * @param {Node} node - Dialog element
-   * @param {Node} main - Main element of the page
-   */
-  var A11yDialog = function (node, main) {
-    main = main || document.querySelector('#main');
-    var that = this;
-    var openers = $$('[data-a11y-dialog-show="' + node.id + '"]');
-    var closers = $$('[data-a11y-dialog-hide]', node)
-      .concat($$('[data-a11y-dialog-hide="' + node.id + '"]'));
-
-    if (node.hasAttribute('aria-hidden')) {
-      this.shown = !JSON.parse(node.getAttribute('aria-hidden'));
+    // Keep an object of listener types mapped to callback functions
+    this._listeners = {
+      show: [],
+      hide: [],
+      destroy: []
+    };
+
+    // Keep a collection of nodes to disable/enable when toggling the dialog
+    this._targets = collect(targets) || getSiblings(this.node);
+
+    // Define the value of the `shown` property based on the initial value of
+    // the `aria-hidden` attribute from the dialog element
+    if (this.node.hasAttribute('aria-hidden')) {
+      this.shown = !JSON.parse(this.node.getAttribute('aria-hidden'));
     }
 
-    this.show = show;
-    this.hide = hide;
-    this.destroy = destroy;
+    // Keep a collection of dialog openers, each of which will be bound a click
+    // event listener to open the dialog
+    this._openers = $$('[data-a11y-dialog-show="' + this.node.id + '"]');
```

Comment:
Sort of. `id` will be undefined, which will be caster as `"undefined"`, which will result in looking for `[data-a11y-dialog-show="undefined"]`, which will return an empty array. At this point, it means there are no openers, and the dialog will be interacted with programmatically.